src/System.IO.Ports/src/System/IO/Ports/SerialStream.cs

```diff
@@ -714,13 +716,7 @@ internal int BytesToWrite
 
                 // prep. for starting event cycle.
                 _eventRunner = new EventLoopRunner(this);
-
-                // We should consider migrating to a Task here rather than simple background Thread
-                // This would let any exceptions be marshalled back to the owner of the SerialPort
-                // Some discussion in GH issue #17666
-                Thread eventLoopThread = new Thread(_eventRunner.SafelyWaitForCommEvent);
-                eventLoopThread.IsBackground = true;
-                eventLoopThread.Start();
+                _waitForComEventTask = Task.Run(() => _eventRunner.SafelyWaitForCommEvent());
```

Comment:
This would be better as:
```C#
_waitForComEventTask = Task.Factory.StartNew(s => ((EventLoopRunner)s).SafelyWaitForCommEvent(), _eventRunner,
    CancellationToken.None, TaskCreationOptions.DenyChildAttach, TaskScheduler.Default);
```
which is the same as using Task.Run except allowing an argument to be passed in to avoid the delegate allocation that results from needing to capture `this` to access `_eventRunner`.